lnwallet/channel.go

```diff
@@ -4140,6 +4150,65 @@ func (lc *LightningChannel) ReceiveNewCommitment(commitSig lnwire.Sig,
 	return nil
 }
 
+// OweCommitment returns a boolean value reflecting whether we need to send
+// out a commitment signature because there are outstanding local updates and/or
+// updates in the local commit tx that aren't reflected in the remote commit tx
+// yet.
+func (lc *LightningChannel) OweCommitment(local bool) bool {
+	lc.RLock()
```

Comment:
Perhaps we should instead obtain the mutex once within `ReceiveNewCommitment`, then either make this a private method, or update the `godoc` to note that it expects the mutex to already be held?

Reply:
It is also called without a lock from `ChannelLink`. I now created a private no-lock version and a public method that wraps it in a lock.

I am wondering whether our channel manipulations are always safe though. Something like this for example:

```
if lc.OweCommitment(true) {
			commitSig, htlcSigs, _, err := lc.SignNextCommitment()
```

I think the channel isn't modified concurrently, but why it the lock there then? Is it a safety precaution? Or for reading channel state in another subsystem?

Reply:
The lock was mostly a safety precaution when the struct was first created just to ensure things were goroutine safe, so all other code added afterwards followed that trend. In the past there were also instances of grabbing live state machines from the peer, and using that to populate RPC information.